### ISSUE TITLE
Integrate learning rate schedulers (CosineAnnealing, StepLR)

### ISSUE THREAD

**Issue by VARUN3WARE** [VARUN3WARE/dml-py#4]
Add support for PyTorch learning rate schedulers in BaseCollaborativeTrainer. Allow users to pass schedulers and automatically step them during training.
